### PR TITLE
feat(option): add support for user loggers

### PR DESCRIPTION
**Pull request by codyoss**
Also add an intenral option that will be used by our clients to extract a logger from the options, or a default logger if one is not specified.

More changes will be needed to make this pass through logger to auth layers in a future iteration.

### CODE REVIEW

File: option/internaloption/internaloption.go [hunk 1]
```diff
@@ -212,3 +215,16 @@ func (w enableNewAuthLibrary) Apply(o *internal.DialSettings) {
 type EmbeddableAdapter struct{}
 
 func (*EmbeddableAdapter) Apply(_ *internal.DialSettings) {}
+
+// GetLogger is a helper for client libraries to extract the from the provided options
```
Comment (quartzmo) [inline option/internaloption/internaloption.go:219]: typo: "the from the"

File: option/option.go [hunk 2]
```diff
@@ -368,3 +369,17 @@ type withUniverseDomain string
 func (w withUniverseDomain) Apply(o *internal.DialSettings) {
 	o.UniverseDomain = string(w)
 }
+
+// WithLogger returns a ClientOption that sets the logger used throughout the
+// client library call stack. If this option is provided it takes precedence
+// over the value set in GOOGLE_SDK_GO_LOGGING_LEVEL. Specifying this option
+// enables logging at the provided loggers configured level.
```
Comment (quartzmo) [inline option/option.go:376]: s/loggers/logger's/

**Review by quartzmo — CHANGES_REQUESTED** [review]

File: option/internaloption/internaloption.go (hunk 1 above)
Comment (codyoss) [reply option/internaloption/internaloption.go]: done

File: option/option.go (hunk 2 above)
Comment (codyoss) [reply option/option.go]: done

File: option/internaloption/internaloption.go [hunk 3]
```diff
@@ -212,3 +215,16 @@ func (w enableNewAuthLibrary) Apply(o *internal.DialSettings) {
 type EmbeddableAdapter struct{}
 
 func (*EmbeddableAdapter) Apply(_ *internal.DialSettings) {}
+
+// GetLogger is a helper for client libraries to extract the [slog.Logger]from
```
Comment (quartzmo) [inline option/internaloption/internaloption.go:219]: typo: s/[slog.Logger]from/[slog.Logger] from/

**Review by quartzmo — APPROVED** [review]

**Pull request merged**